caffe-distri/src/main/cpp/CaffeNet.cpp

```diff
@@ -531,7 +531,10 @@ int CaffeNet<Dtype>::getInitIter(int solver_index) {
         return root_solver_->iter();
     else {
         CHECK(syncs_[solver_index]);
-        return syncs_[solver_index]->solver()->iter();
+	if (solver_index == 0)
```

Comment:
We will need some logic to handle CPU mode. 
https://github.com/yahoo/CaffeOnSpark/blob/master/caffe-distri/src/main/cpp/CaffeNet.cpp#L208-L218

`if (solver_index == 0 || solver_mode_ == Caffe::CPU)`